riff-cli/cmd/publish.go

```diff
@@ -157,11 +158,16 @@ func doPost(url string, publishOptions publishOptions) (string, error) {
 	if err != nil {
 		return "", err
 	}
-
 	defer resp.Body.Close()
+
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return "", err
 	}
-	return string(body), nil
+	if 200 <= resp.StatusCode && resp.StatusCode < 400 {
+		return string(body), nil
+	} else {
+		return "", fmt.Errorf("Gateway responded with code %v: %v", resp.StatusCode, string(body))
```

Comment:
"Gateway" could mean lots of things in a network. Perhaps change to "HTTP gateway".

Reply:
Will do.

OK to merge afterwards?